File: src/com/brianc/layout/Fragment.java
package com.brianc.layout;

import java.awt.font.TextLayout;
import java.util.BitSet;

import com.brianc.style.StyledNode;

public class Fragment {
	StyledNode nodeRef;
	InlineBox box;
	int startPos;
	int endPos;
	TextLayout lineLayout;
	BitSet flags;

	public static final int FIRST_FOR_ELEMENT = 1;
	public static final int LAST_FOR_ELEMENT = 2;

	public Fragment(InlineBox box, int startPos, int endPos, TextLayout lineLayout) {
		this.box = box;
		this.nodeRef = box.getStyledNode();
		this.lineLayout = lineLayout;
		this.startPos = startPos;
		this.endPos = endPos;
		this.flags = new BitSet();
	}
	
	public BitSet getFlags() {
		return flags;
	}

	public TextLayout getLayout() {
		return lineLayout;
	}
}

File: src/com/brianc/layout/Fragment.kt
package com.brianc.layout

import java.awt.font.TextLayout
import java.util.BitSet

import com.brianc.style.StyledNode

class Fragment(internal var box: InlineBox, internal var startPos: Int, internal var endPos: Int, lineLayout: TextLayout) {
    internal var nodeRef: StyledNode
    var layout: TextLayout
        internal set
    var flags: BitSet
        internal set

    init {
        this.nodeRef = box.styledNode
        this.layout = lineLayout
        this.flags = BitSet()
    }

    companion object {

        val FIRST_FOR_ELEMENT = 1
        val LAST_FOR_ELEMENT = 2
    }
}
